lib/rack/request.rb

```diff
@@ -207,8 +219,42 @@ def scheme
         end
       end
 
+      # The authority of the incoming reuqest as defined by RFC2976.
+      # https://tools.ietf.org/html/rfc3986#section-3.2
+      #
+      # In HTTP/1, this is the `host` header.
+      # In HTTP/2, this is the `:authority` pseudo-header.
       def authority
-        get_header(SERVER_NAME) + ':' + get_header(SERVER_PORT)
+        forwarded_authority || host_authority || server_authority
+      end
+
+      # The authority as defined by the `SERVER_NAME`/`SERVER_ADDR` and
+      # `SERVER_PORT` variables.
+      def server_authority
+        host = self.server_name
+        port = self.server_port
+
+        if host
+          if port
+            return "#{host}:#{port}"
+          else
+            return host
+          end
+        end
+      end
+
+      def server_name
+        if name = get_header(SERVER_NAME)
+          return name
```

Comment:
In https://github.com/rack/rack/pull/1557#issuecomment-582611953, you stated `I'm fine with implicit return when the cyclomatic complexity of a function is 1, or the branch depth is 1.`  To keep at least a reasonably consistent style, can we avoid explicit returns in cases where the branch depth is <= 1 (I assume the branch depth is 1 here, and outside of the if the branch depth is 0).  Ditto for the rest of the file.